Feature/#3 liquibase

kos24 opened the pull request:
#3: addded liquibase changesets

erenkov committed:
#4: add entity: task, release, taskkRelease. pom.xml changed

erenkov committed:
#4: add entity: task, release, taskkRelease. pom.xml changed +

erenkov committed:
add Composite Primary Keys to TaskReleaseEntity

sifdragon committed:
Added enum entities (#7)

* Added enum entities

* Added TaskType enum

Co-authored-by: IlnazG <wistfuldr@mail.com>

sifdragon committed:
Merge pull request #8 from erenkov/subtask/#4-gilmutdinov-2

Updated enum's + added converters for enum's

Commit:
Updated enum's + added converters for enum's

kos24 committed:
#4: added entities

kos24 committed:
Merge branch 'feature/#4-entity' into subtask/#4-rebrin

kos24 committed:
Merge pull request #9 from erenkov/subtask/#4-rebrin

Subtask/#4 rebrin

kos24 committed:
#3: added liquibase ddl script

kos24 committed:
#3: changed changelog-1.0

kos24 committed:
#3: minor fixes in ddl script

hotabsc commented on src/main/java/com/developing/simbir_product/converter/ProjectStatusConverter.java:
```diff
@@ -0,0 +1,33 @@
+                return ProjectStatus.CLOSED;
+
+            default:
+                throw new IllegalArgumentException("ShortName [" + shortName
```
Советую использовать String.format()

kos24 replied:
Исправил в задаче #4, так как в ней актуальные конвертеры

hotabsc commented on src/main/java/com/developing/simbir_product/entity/TaskEntity.java:
```diff
@@ -0,0 +1,180 @@
+import java.util.UUID;
+
+@Entity
+@Table(name = "TASK")
```
почему большими буквами?)

kos24 replied:
чтобы понимать, что это названия таблиц в БД) обычно прописывают маленькими?

hotabsc replied:
что-то я совсем не видела чтобы прописывали большими

hotabsc commented on src/main/java/com/developing/simbir_product/service/ReleaseService.java:
```diff
@@ -0,0 +1,10 @@
+
+public interface ReleaseService {
+
+    private ReleaseResponseDto getById(UUID is) {
```
скорее всего `id`, а не `is`

hotabsc replied:
почему методы private и уже с реализацией?

kos24 replied:
классы сервисного слоя попали в эту ветку по ошибке, у нас будет отдельный PR #11 по сервисам

hotabsc commented on src/main/java/com/developing/simbir_product/service/impl/ReleaseServiceImpl.java:
```diff
@@ -0,0 +1,14 @@
+
+    @Autowired
+    private ReleaseRepository releaseRepository;
+
```
в *ServiceImpl должны быть переопределены методы интерфейса

kos24 replied:
классы сервисного слоя попали в эту ветку по ошибке (это не актуальная реализация), у нас будет отдельный PR #11 по сервисам

hotabsc commented on src/main/resources/db/changelog/migrations/changelog-1.0.yaml:
```diff
@@ -0,0 +1,364 @@
+            remarks: Информация о задачах
+            tableName: task
+  - changeSet:
+      id: 1619638368613-2
```
по какому принципу генерировался id-ик?)

kos24 replied:
сгенерировал автоматически, когда формировал changelog, нужно исправить на более осмысленный?

hotabsc replied:
не принципиально

kos24 committed:
#3: changed table name for USR and changed field to pass

Nikitinmn42 committed:
#3: Added user_number column to usr table

kos24 committed:
# This is a combination of 3 commits.
# This is the 1st commit message:

#3 added FK for user_task_history

# This is the commit message #2:

#3: changed table name for task_release_history

# This is the commit message #3:

#3:changed liquibase: rename table task_release_history, added FK

kos24 committed:
#3 added FK for user_task_history

kos24 committed:
#3: changed table name for task_release_history

kos24 committed:
#3:changed liquibase: rename table task_release_history, added FK

kos24 committed:
#3: added surrogate keys for User_Task_History and Task_Release_History

kos24 committed:
Merge branch 'bugfix/#3-add-usernumber' of https://github.com/erenkov/project-manager into bugfix/#3-add-usernumber

Nikitinmn42 committed:
Merge pull request #19 from erenkov/bugfix/#3-add-usernumber

#3: Added user_number column to usr table
